Angular core module

craigburke opened the pull request.

Schlogen commented:
Give me time to review before merging

craigburke commented:
@Schlogen sounds good. Let me know what you think.

Schlogen commented on profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js:
```diff
@@ -0,0 +1,14 @@
+//= wrapped
```
Should this have the require for /angular/angular ?

craigburke replied:
I usually put the require for angular in core module with all the other dependencies but I don't have a strong opinion on that either way. Let me know if you want me to move that to `app.js`

Schlogen commented on profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js:
```diff
@@ -0,0 +1,14 @@
+//= require /@grails.codegen.defaultPackage.path@/core/core
+//= require /@grails.codegen.defaultPackage.path@/home/home
+//= require_self
+//= require_tree services
```
Given the main app typically wont have angular controllers, etc directly; Does it make sense to remove the requires for controllers, directives, etc ?

craigburke replied:
Missed this comment the first time. Yeah, absolutely.

craigburke commented:
After some very thoughtful feedback from @Schlogen I made the following changes:

* Renamed the home module to index
* Added optional arguments for DomainServiceFactory so that arguments to $resource can be overridden for individual services.
*  Check for DomainServiceFactory when `create-ng-domain` is called and fall back to using $resource directly if it's not found

Schlogen commented on profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/core/services/DomainServiceFactory.js:
```diff
@@ -0,0 +1,22 @@
+        return $resource(
+            url,
+            paramDefaults || null,
+            actions || {"update": {method: "PUT"}, "list": {method: "GET", isArray: true}},
```
For this param I would use extends. That way if the user only wants to add a method, they can keep the defaults.

craigburke replied:
That makes sense.
